rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -332,8 +333,11 @@ impl LightFetch {
 			// match them with their numbers for easy sorting later.
 			let bit_combos = filter.bloom_possibilities();
 			let receipts_futures: Vec<_> = self.client.ancestry_iter(filter.to_block)
-				.take_while(|ref hdr| BlockId::Number(hdr.number()) != filter.from_block)
-				.take_while(|ref hdr| BlockId::Hash(hdr.hash()) != filter.from_block)
+				.take_while(|ref hdr|{
+					BlockId::Hash(hdr.hash()) != filter.from_block
+					&& BlockId::Number(hdr.number()) != filter.from_block
+				})
+				.chain(::std::iter::once(self.client.block_header(filter.from_block).expect("checked before")))
 				.filter(|ref hdr| {
 					let hdr_bloom = hdr.log_bloom();
 					bit_combos.iter().find(|&bloom| hdr_bloom & *bloom == *bloom).is_some()
```

Comment:
not related to this pr, but it's more optimal to use `contains_bloom`

Reply:
Yes, this kind of logic should definitely use dedicated function.